jobspec/parse_task.go

```diff
@@ -169,6 +169,10 @@ func parseTask(item *ast.ObjectItem, keys []string) (*api.Task, error) {
 		if err := hcl.DecodeObject(&m, i.Val); err != nil {
 			return nil, err
 		}
+		if cfg.StagePublishBaseDir != "" && cfg.MountDir != "" &&
+			strings.Contains(cfg.StagePublishBaseDir, cfg.MountDir) {
+			return nil, fmt.Errorf("csi_plugin -> Expected stage_publish_base_dir to not be a subdirectory of mount_dir")
+		}
```

Comment:
There's two problems here:
* This isn't the right place to validate this. This package is the HCL1 jobspec parsing. So HCL2 isn't covered and neither are any job registration requests that come in through the API. You'll want to do this when we validate the [`CSIPluginConfig`](https://github.com/hashicorp/nomad/blob/71ea1deda707b706c36da7567caba7721e51db19/nomad/structs/structs.go#L7935-L7946) struct so that any job registration hits this check.
* This only checks if the path string for the staging publish base dir _contains_ the string path for the mount dir, not that they're relative to each other. These could be hypothetically non-overlapping values like `/local/csi` and `/csi` and that would fail.

Reply:
Ah of course - will address this today. Thanks